### DISCUSSION TITLE
How to format selected filters in search results

### DISCUSSION THREAD

**Discussion by egwrk** [microsoft-search/pnp-modern-search#3775] [Q&A]
I have the "Show selected filters" option enabled in Search Results. However, the default format does not seem to be very user-friendly. Is there a way to change the format so that it has the desired layout below? 

I tired searching through the templates here: https://github.com/microsoft-search/pnp-modern-search-layouts/tree/main/Results
However, there are not many options and none seem to have what I'm looking for. 

![image](https://github.com/microsoft-search/pnp-modern-search/assets/161059375/9b4b93e4-9d86-46b8-bd80-54a8c11caa05)

**Comment by wobba** [comment]
You need a developer to change how the web part works to add this kind of UX.

**Reply by egwrk** [reply]
Thank you @wobba! My team is also looking to adjust the results so that if a filter is selected, or results match the search term, to bold the matching text. Example: 'Cyber Security' filter is selected which makes 'Cyber Security' bolded text in the results. We would also like to have ellipses to break up long text. Would these two requirements above also need a developer? 

![image](https://github.com/microsoft-search/pnp-modern-search/assets/161059375/4e5ad2b4-fa8f-4e4a-a519-ead12e30734a)

**Reply by wobba** [reply]
If the returned hithighlighted summary does not work for you, then creating something yourself via code might be easier compared to handlbar function trickery.

**Reply by egwrk** [reply]
@wobba, the hithighlighted summary did not work, unfortunately. Do you know if changing the UX to the example above would require an extensive code change from the developers?

**Reply by kasperbolarsen** [reply]
Regarding the initial post: I would have a hard time accepting the desired layout as it is. If we don't show the name of the filter the end user will have a very hard time figuring out which filter is used when we have multiple filters available ( and the values might be similar or identical across the filters)

**Reply by egwrk** [reply]
Our UX team has updated the desired layout to the below. Would this design require an extensive code change from the developers on my team?

![SkillsPOC_Design](https://github.com/user-attachments/assets/3dfe41ca-274d-4f0a-80ce-7c07bc09d155)

**Reply by kasperbolarsen** [reply]
Not at all, if your devs have a skill set like mine it should not take more than approx 3-400 hours to complete. 



Honest, there is no way we can estimate how long it will take your devs to complete anything based on a drawing.
I have to ask the devs, they're the only ones that might have a ballpark guestimate
